src/charm.py

```diff
@@ -205,6 +197,44 @@ def _check_kf_profiles(self, interfaces):
 
         return kf_profiles
 
+    def main(self, event) -> None:
+        """Main entry point for the Charm."""
+        try:
+            self._check_container_connection()
+            self._check_model_name()
+            self._check_leader()
+            interfaces = self._get_interfaces()
+            kf_profiles = self._check_kf_profiles(interfaces)
+        except CheckFailed as e:
+            self.model.unit.status = e.status
+            return
+        try:
+            self.unit.status = MaintenanceStatus("Creating k8s resources")
+            self.k8s_resource_handler.apply()
+        except ApiError:
+            self.logger.error(traceback.format_exc())
+            self.unit.status = BlockedStatus("kubernetes resource creation failed")
+        self.handle_ingress(interfaces)
+        kf_profiles = list(kf_profiles.get_data().values())[0]
+        self.profiles_service = kf_profiles["service-name"]
+        try:
+            self._update_layer()
+        except CheckFailed as e:
+            self.model.unit.status = e.status
+            return
+        self.model.unit.status = ActiveStatus()
+
+    def _on_remove(self, event):
+        self.unit.status = MaintenanceStatus("Removing k8s resources")
```

Comment:
AFAIK, we do not handle status on remove event handlers.

Reply:
Pietro suggested to set status in this event for better unittesting.